components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -896,222 +895,65 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
 
           !.......................
           ! collection of droplets
-
-          ! here we multiply rates by air density, air density fallspeed correction
-          ! factor, and collection efficiency since these parameters are not
-          ! included in lookup table calculations
-          ! for T < 273.15, assume collected cloud water is instantly frozen
-          ! note 'f1pr' values are normalized, so we need to multiply by N
-
-          if (qitot_incld(i,k).ge.qsmall .and. qc_incld(i,k).ge.qsmall .and. t(i,k).le.zerodegc) then
-             qccol = rhofaci(i,k)*f1pr04*qc_incld(i,k)*eci*rho(i,k)*nitot_incld(i,k)
-             nccol = rhofaci(i,k)*f1pr04*nc_incld(i,k)*eci*rho(i,k)*nitot_incld(i,k)
-          endif
-
-          ! for T > 273.15, assume cloud water is collected and shed as rain drops
-
-          if (qitot_incld(i,k).ge.qsmall .and. qc_incld(i,k).ge.qsmall .and. t(i,k).gt.zerodegc) then
-             ! sink for cloud water mass and number, note qcshed is source for rain mass
-             qcshd = rhofaci(i,k)*f1pr04*qc_incld(i,k)*eci*rho(i,k)*nitot_incld(i,k)
-             nccol = rhofaci(i,k)*f1pr04*nc_incld(i,k)*eci*rho(i,k)*nitot_incld(i,k)
-             ! source for rain number, assume 1 mm drops are shed
-             ncshdc = qcshd*1.923e+6_rtype
-          endif
+          call ice_cld_liquid_collection(rho(i,k),t(i,k),rhofaci(i,k),&
```

Comment:
At the risk of being fussy, I would prefer 'cldliq' instead of 'cld_liquid' here because the "_" makes it look like a 3rd process: ice->cld->liquid. Also, I like short names so long as they are clear... cldliquid would be fine with me too though.

Reply:
I agree, the code has been updated accordingly.